### PR TITLE
implement permissions

### PR DESCRIPTION
**Pull request by muckmuck96**
Implement permissions and predescribed some perms to the modules controller. Please check if the perms are set correctly. :)

### CODE REVIEW

File: src/modules/modules.controller.ts [hunk 1]
```diff
@@ -16,12 +18,14 @@ import { TimeSlotsService } from './timeslots/time-slots.service';
...
   constructor(private modulesService: ModulesService, private timeSlotService: TimeSlotsService) {}
 
+  @UserTypes(UserEnum.ADMIN)
   @ApiCommonResponse({ type: ModuleDTO })
   @Post('/')
   async create(@Request() req: RequestWithUser, @Body() module: CreateModuleDTO): Promise<ModuleDTO> {
     return await this.modulesService.create(module);
   }
 
+  @UserTypes(UserEnum.TEACHER, UserEnum.TEACHING_STUDENT)
```
Comment (koanbockFH) [inline src/modules/modules.controller.ts]: I don't think a teacher or student should be able to edit the module - that is Admin only

File: src/modules/modules.controller.ts [hunk 2]
```diff
@@ -34,6 +38,7 @@ export class ModulesController {
     return await this.timeSlotService.getById(id);
   }
 
+  @UserTypes(UserEnum.TEACHER, UserEnum.TEACHING_STUDENT)
```
Comment (koanbockFH) [inline src/modules/modules.controller.ts]: Admin only

File: src/modules/modules.controller.ts [hunk 3]
```diff
@@ -51,17 +56,20 @@ export class ModulesController {
     return await this.modulesService.getAllByUser(req.user.id);
   }
 
+  @UserTypes(UserEnum.ADMIN)
   @Delete('/:id')
   async delete(@Request() req: RequestWithUser, @Param('id') id: number): Promise<void> {
     await this.modulesService.delete(id);
   }
 
+  @UserTypes(UserEnum.TEACHER, UserEnum.TEACHING_STUDENT)
```
Comment (koanbockFH) [inline src/modules/modules.controller.ts:65]: add admin

File: src/modules/modules.controller.ts [hunk 4]
```diff
@@ -51,17 +56,20 @@ export class ModulesController {
...
   @ApiCommonResponse({ type: TimeSlotDTO })
   @Post('/:moduleId/timeslot')
   async createTimeSlot(@Param('moduleId') moduleId: number, @Body() timeSlot: CreateTimeSlotDTO): Promise<TimeSlotDTO> {
     return await this.timeSlotService.create(moduleId, timeSlot);
   }
 
+  @UserTypes(UserEnum.TEACHER, UserEnum.TEACHING_STUDENT)
   @ApiCommonResponse({ type: TimeSlotDTO })
   @Put('/:moduleId/timeslot')
   async editTimeSlot(@Param('moduleId') moduleId: number, @Body() timeSlot: TimeSlotDTO): Promise<TimeSlotDTO> {
```
Comment (koanbockFH) [inline src/modules/modules.controller.ts:75]: add admin

**Review by koanbockFH — CHANGES_REQUESTED** [review]

**Comment by koanbockFH** [comment]
I can't access GetModules while logged in as student or as teacher  -- only works as admin

File: src/modules/modules.controller.ts (hunk 3 above)
Comment (muckmuck96) [reply src/modules/modules.controller.ts:65]: Admin has always rights, so no need to add him here

File: src/modules/modules.controller.ts (hunk 4 above)
Comment (muckmuck96) [reply src/modules/modules.controller.ts:75]: Admin has always rights, so no need to add him here

Comment (koanbockFH) [reply src/modules/modules.controller.ts:75]: ah ok didn't realize that

**Comment by muckmuck96** [comment]
So is this a should be or should not be? Or what do you mean by this? There is only the /api/v1/module GET request and you get only the modules you are in, as you made.

**Comment by koanbockFH** [comment]
Yeah the thing is if I am logged in as a Student or Teacher - I get a forbidden response but I should get the modules (well in this case an empty list since they don't have modules)

**Comment by muckmuck96** [comment]
Oh ok, I'll have a look at this.

**Comment by koanbockFH** [comment]
this also is the case with other requests that do not have any UserType decorator on them (so basically free for all requests)

**Review by koanbockFH — APPROVED** [review]
LGTM

**Pull request merged**